sqlccl: log cancelled stmts in TestExplainGist

yuzefovich opened the pull request:
We just saw a test failure that wasn't reproducible by simply rerunning the test with the seed that was reported on the issue. This is the case since we have some non-determinism in sqlsmith that we haven't fully fleshed out. This test is aware of the non-determinism, so it attempts to keep track of the successful stmts which are printed out on a test failure, but in this case this was also insufficient. The query that hit an internal error attempts to reference a column that isn't in the CREATE TABLE, so it must have been added as a schema change, but we don't see it in the log of successful statements. TestExplainGist executes all stmts with 0.1s statement timeout (because initially it was envisioned as the stress of plan-gist logic, so we wanted to have some change in the DB state, but we didn't care much whether all changes would actually apply), so many schema changes are probably canceled before they can complete. This commit makes it so that we now also track all such cancelled statements which we print out in the commented out form on a test failure.

Additionally, this commit skips a couple of known test failures that have reproductions to reduce the noise.

Informs: #149552.
Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/149637)
<!-- Reviewable:end -->

yuzefovich committed:
sqlccl: log cancelled stmts in TestExplainGist

We just saw a test failure that wasn't reproducible by simply rerunning
the test with the seed that was reported on the issue. This is the case
since we have some non-determinism in sqlsmith that we haven't fully
fleshed out. This test is aware of the non-determinism, so it attempts
to keep track of the successful stmts which are printed out on a test
failure, but in this case this was also insufficient. The query that hit
an internal error attempts to reference a column that isn't in the
CREATE TABLE, so it must have been added as a schema change, but we
don't see it in the log of successful statements. TestExplainGist
executes all stmts with 0.1s statement timeout (because initially it was
envisioned as the stress of plan-gist logic, so we wanted to have some
change in the DB state, but we didn't care much whether all changes
would actually apply), so many schema changes are probably canceled
before they can complete. This commit makes it so that we now also
track all such cancelled statements which we print out in the commented
out form on a test failure.

Additionally, this commit skips a couple of known test failures that
have reproductions to reduce the noise.

Release note: None

michae2 commented:
I bet this will make @mw5h happy!

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/149637#-:-OUec9-S4GskV_RXctqd:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mw5h)



<!-- Sent from Reviewable.io -->

mw5h commented:
Thanks for doing this. This drove me crazy yesterday!

:lgtm:
